generators/client/templates/react/src/main/webapp/app/shared/error/error-boundary.tsx.ejs

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+interface IErrorBoundaryProps {
+  readonly children: JSX.Element;
+}
+
+interface IErrorBoundaryState {
+  readonly error: any;
+  readonly errorInfo: any;
+}
+
+class ErrorBoundary extends React.Component<IErrorBoundaryProps, IErrorBoundaryState> {
+  readonly state: IErrorBoundaryState = { error: undefined, errorInfo: undefined };
+
+  componentDidCatch(error, errorInfo) {
+    this.setState({
+      error,
+      errorInfo
+    });
+  }
+
+  render() {
+    if (this.state.errorInfo) {
```

Comment:
I think deconstructing the state and prop here might be cleaner so that we dont have to do so many this.state and this.prop below

Reply:
We access both error and errorInfo twice in render and I think it should be fine. We use props only once in render. IMO, it's not that complex to move to local variables, however, if you still feel it should be done, I can refactor this

Reply:
Its just about neatness, having `const { error, errorInfo } = this.state;` and using just `error` and `errorInfo` it in render seems a bit more nicer, but if you do not want to do it its fine as well